src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs

```diff
@@ -104,9 +104,7 @@ private TagSource GetOrCreateTagSource(ITextView textViewOpt, ITextBuffer subjec
             if (!this.TryRetrieveTagSource(textViewOpt, subjectBuffer, out var tagSource))
             {
                 tagSource = new TagSource(textViewOpt, subjectBuffer, this, AsyncListener, _notificationService);
-
                 this.StoreTagSource(textViewOpt, subjectBuffer, tagSource);
-                tagSource.Disposed += (s, e) => this.RemoveTagSource(textViewOpt, subjectBuffer);
```

Comment:
instead of hearing when the tagSource is disposed, teh tag source just removes itself from teh TaggerProvider in disposal.  See https://github.com/dotnet/roslyn/pull/52790/files#diff-e4bbaf85df911375a936267fb99cb9cf629bc34ab33fb036a0ff3303e9192885R159